node-manager/src/nodemanager.rs

```diff
@@ -502,33 +508,45 @@ impl NodeManager {
     }
 
     #[wasm_bindgen]
-    // todo change return to MutinyInvoice
     pub async fn pay_invoice(
         &self,
         from_node: String,
         invoice_str: String,
-    ) -> Result<(), MutinyJsError> {
+    ) -> Result<MutinyInvoice, MutinyJsError> {
         let nodes = self.nodes.lock().await;
         let node = nodes.get(from_node.as_str()).unwrap();
 
         let invoice =
             Invoice::from_str(invoice_str.as_str()).map_err(|_| MutinyJsError::InvoiceInvalid)?;
-        let pay_result = node.invoice_payer.pay_invoice(&invoice);
 
-        match pay_result {
-            Ok(_) => Ok(info!("Payment successful!")),
-            Err(_) => Err(MutinyJsError::RoutingFailed),
+        // todo ensure payment hash is unique
+        let payment_hash = PaymentHash(invoice.payment_hash().into_32());
+
+        let _ = node.invoice_payer.pay_invoice(&invoice)?;
+
+        info!("Payment successful!");
+
+        match node.persister.read_payment_info(payment_hash, false) {
+            Some(payment_info) => {
+                let mut mutiny_invoice = MutinyInvoice::from(invoice);
+
+                mutiny_invoice.is_send = true;
+                mutiny_invoice.paid = true;
+                mutiny_invoice.fees_paid = payment_info.fee_paid_msat;
+
+                Ok(mutiny_invoice)
+            }
+            None => Err(MutinyJsError::ReadError),
```

Comment:
Unfortunately this won't work. the invoice payer is async so it really only indicates that a payment is in progreess.

Reply:
In order to have a blocking call, we need a channel between event handler and this that blocks on waiting for the event handler to emit a payment success event.

Reply:
reverted these changes as they'll be addresses in your PR anyways